services/requester/batch_tx_pool.go

```diff
@@ -151,20 +143,46 @@ func (t *BatchTxPool) Add(
 	// For all 3 cases, we record the activity time for the next
 	// transactions that might come from the same EOA.
 	// [X] is equal to the configured `TxBatchInterval` duration.
-	lastActivityTime, found := t.eoaActivity.Get(from)
-	if !found || time.Since(lastActivityTime) > t.config.TxBatchInterval {
-		select {
-		case <-t.done:
-			return fmt.Errorf("the server is shutting down")
-		default:
-			t.txChan <- hexEncodedTx
+	lastActivity, found := t.eoaActivity.Get(from)
```

Comment:
maybe I missed something, but why do we have to index activities (tx hashes) by address? Isn't the `from` address part of the tx hash?

Can we create a separate `recentTxHashes` LRU for deduplication?

Reply:
I have removed this functionality altogether, with regards to https://github.com/onflow/flow-evm-gateway/pull/852#discussion_r2229103641 .